src/startupscriptgenerator/src/common/setupEnvironment.go

```diff
@@ -29,6 +30,14 @@ func SetupEnv(script string) {
 }
 
 func GetSetupScript(platformName string, version string, installationDir string) string {
+	dowloadSentinelFileName := ".oryx-sentinel"
+	sentinelFilePath := path.Join(installationDir, dowloadSentinelFileName)
```

Comment:
wondering what is sentinel file used for runtime image?

Reply:
It is basically a marker file to indicate that the earlier download of the platform runtime was indeed successful. If you notice the generated script for the file download, we create this file only after the download has completed. This is just a gate to handle scenarios where users might stop the download of the runtime midway resulting in an inconsistent state of the runtime.